go/ai/generate.go

```diff
@@ -531,7 +582,7 @@ func validResponse(ctx context.Context, resp *ModelResponse) (*Message, error) {
 // validMessage will validate the message against the expected schema.
 // It will return an error if it does not match, otherwise it will return a message with JSON content and type.
 func validMessage(m *Message, output *ModelOutputConfig) (*Message, error) {
-	if output != nil && output.Format == string(OutputFormatJSON) {
+	if output != nil && output.Format == string(OutputFormatJSON) && output.Constrained {
```

Comment:
I wanted to check if this condition makes sense at this place. From what I understand, we don't require to validate the response message if no output schema is provided.